packages/app-runtime/src/modules/runtimeEvents/IdentityDeletionProcessStatusChangedModule.ts

```diff
@@ -19,25 +21,44 @@ export class IdentityDeletionProcessStatusChangedModule extends AppRuntimeModule
     private async handleIdentityDeletionProcessStatusChanged(event: IdentityDeletionProcessStatusChangedEvent) {
         const identityDeletionProcess = event.data;
 
+        if (!identityDeletionProcess) {
+            const services = await this.runtime.getServices(event.eventTargetAddress);
+            const identityDeletionProcesses = await services.transportServices.identityDeletionProcesses.getIdentityDeletionProcesses();
+
+            const approvedIdentityDeletionProcess = identityDeletionProcesses.value.filter((idp) => idp.status === IdentityDeletionProcessStatus.Approved).at(0);
+            const deletionDate = approvedIdentityDeletionProcess?.gracePeriodEndsAt ? CoreDate.from(approvedIdentityDeletionProcess.gracePeriodEndsAt) : undefined;
+
+            await this.updateLocalAccountDeletionDate(event.eventTargetAddress, deletionDate, true);
+            return;
+        }
```

Comment:
```suggestion
        let publishEvent = false;
        let identityDeletionProcess = event.data;

        if (!identityDeletionProcess) {
            const services = await this.runtime.getServices(event.eventTargetAddress);
            const identityDeletionProcesses = await services.transportServices.identityDeletionProcesses.getIdentityDeletionProcesses();
            const identityDeletionProcess = identityDeletionProcesses.value.at(0)!;
            publishEvent = true;
        }
```

Reply:
What about this? To me it seems simpler.

Reply:
But we save the filtering step before... Maybe filtering is better than the switch logic perfomance-wise, but I find it the other way around a bit easier to comprehend.

Reply:
ACtually the filtering does more than the switch as the `default:` is empty.